library/functions.lisp

```diff
@@ -44,6 +48,17 @@
       (lisp :a (str item) (cl:format cl:t "~A: ~A~%" str item))
       Unit))
 
+  (declare print ((Into :a String) => :a -> Unit))
+  (define print
+    "Print the String representation of an item to `cl:*standard-output*`."
+    (compose trace into))
+
+  (declare and-return ((:a -> :b) -> :a -> :a))
+  (define (and-return func item)
```

Comment:
Is there a use case for this? Not sure this is a useful enough utility to graduate to the standard library.

In any case, I'd probably call it `intercept` or something and reverse the argument order.

Reply:
The main use case is really to implement a `debug-print`, so that, when testing code, once can just wrap an expression with `(and-return print (...))` or `(debug-print (...))` and the code will continue to work as expected. However, I'll remove this, since it is easily implemented by the user when needed.